data/test/scenarios/test_movetype.cfg

```diff
@@ -332,29 +332,164 @@
+#####
 {MOVETYPE_MOVE_SCEN test_elf_slow_cave_movement (Elvish Archer) (1,1,2,2,4,2,2,4,1,1,1,3,1) {ORC_COSTS} {SET_MOVE_COSTS cave=4}}
 
-# changing the vision costs shouldn't affect the movement costs
+#####
+# API(s) being tested: [effect]apply_to=vision_costs
+##
+# Expected end state:
+# The Elvish Archer (alice) has the expected movement costs per terrain based on its default movetype.
+# The Orcish Grunt (bob) has the expected movement costs per terrain based on its default movetype.
+# Changing the vision costs doesn't affect the movement costs.
+#####
 {MOVETYPE_MOVE_SCEN test_elf_longsighted_movement (Elvish Archer) (1,1,2,2,3,2,2,3,1,1,1,3,1) {ORC_COSTS} {SET_VISION_COSTS cave=1}}
+#####
+# API(s) being tested: [effect]apply_to=vision_costs
+##
+# Expected end state:
+# The Orcish Grunt (alice) has the expected movement costs per terrain based on its default movetype.
+# The Orcish Grunt (bob) has the expected movement costs per terrain based on its default movetype.
+# Changing the vision costs doesn't affect the movement costs.
+#####
 {MOVETYPE_MOVE_SCEN test_orc_longsighted_movement (Orcish Grunt) {ORC_COSTS} {ORC_COSTS} {SET_VISION_COSTS cave=1}}
-{MOVETYPE_MOVE_SCEN test_elf_longsighted_fast_cave_movement (Elvish Archer) (1,1,2,2,2,2,2,2,1,1,1,3,1) {ORC_COSTS} ({SET_VISION_COSTS hills=1}{SET_MOVE_COSTS cave=1})}
-{MOVETYPE_MOVE_SCEN test_elf_fast_cave_longsighted_movement (Elvish Archer) (1,1,2,2,2,2,2,2,1,1,1,3,1) {ORC_COSTS} ({SET_MOVE_COSTS cave=1}{SET_VISION_COSTS hills=1})}
+#####
+# API(s) being tested: [effect]apply_to=vision_costs,[effect]apply_to=movement_costs
+##
+# Expected end state:
+# The Elvish Archer (alice) has the expected movement costs per terrain based on its modified movetype with the movement cost of cave set to 1.
+# The Orcish Grunt (bob) has the expected movement costs per terrain based on its default movetype.
+# Changing the vision costs doesn't affect the movement costs.
+#####
+{MOVETYPE_MOVE_SCEN test_elf_longsighted_fast_cave_movement (Elvish Archer) (1,1,2,2,2,2,2,2,1,1,1,3,1) {ORC_COSTS} (
+    {SET_VISION_COSTS hills=1}
+    {SET_MOVE_COSTS cave=1}
+)}
+#####
+# API(s) being tested: [effect]apply_to=vision_costs,[effect]apply_to=movement_costs
+##
+# Expected end state:
+# The Elvish Archer (alice) has the expected movement costs per terrain based on its modified movetype with the movement cost of cave set to 1.
+# The Orcish Grunt (bob) has the expected movement costs per terrain based on its default movetype.
+# Changing the vision costs doesn't affect the movement costs.
+# Changing the sequence of applying changes to movement cost and vision cost does not affect the outcome.
+#####
+{MOVETYPE_MOVE_SCEN test_elf_fast_cave_longsighted_movement (Elvish Archer) (1,1,2,2,2,2,2,2,1,1,1,3,1) {ORC_COSTS} (
+    {SET_MOVE_COSTS cave=1}
+    {SET_VISION_COSTS hills=1}
+)}
 
+#####
+# API(s) being tested: N/A
```

Comment:
All of the `MOVETYPE_VISION_SCEN` tests in this file are also testing `[store_locations][filter_vision]`.

Reply:
This one is unresolved. It might have got skipped over because it looked like the `MOVETYPE_MOVE_SCEN` comment, which was resolved.